### PR TITLE
f/6112 f/8245 Security updates for Rudder SDK

### PR DESCRIPTION
**Pull request by rohan-epifi**
1. Created encrypted shared pref manager
2. Encrypted db
3. Enabled certificate transparency for both Data & Control plane

### Solves/Fixes:
_https://monorail.pointz.in/p/fi-app/issues/detail?id=6112_
_https://monorail.pointz.in/p/fi-app/issues/detail?id=8245_

### Retrospection
N.A.

### Approach:
Removed dependency from maven rudder library.
Added changes in Rudder SDK & imported custom Rudder aar.

### Review checklist
- [ ] Add any important items for review checklist for the reviewer. They can check off each item when they're done. Examples:
- [ ] All tests passing
- [ ] Added relevant documentation

### Screenshots/GIFs (Of UI changes)
N.A.

### CODE REVIEW

File: core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java [hunk 1]
```diff
@@ -86,6 +95,7 @@
 
             // 4. initiate RudderServerConfigManager
             RudderLogger.logDebug("EventRepository: constructor: Initiating RudderServerConfigManager");
+            SQLiteDatabase.loadLibs(_application);
```
Comment (saran-epifi) [inline core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java:98]: Why is this needed?

File: core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java [hunk 2]
```diff
@@ -405,6 +415,17 @@ private String getPayloadFromMessages(ArrayList<Integer> messageIds, ArrayList<S
             URL url = new URL(dataPlaneEndPoint);
             // get connection object
             HttpURLConnection httpConnection = (HttpURLConnection) url.openConnection();
+            // enable certificate transparency
+            if (httpConnection instanceof HttpsURLConnection) {
+                HttpsURLConnection httpsConnection = (HttpsURLConnection) httpConnection;
+                if (hostnameVerifier == null) {
+                    hostnameVerifier = new CTHostnameVerifierBuilder(httpsConnection.getHostnameVerifier())
+                            .includeHost(url.getHost())
+                            .setLogger(new BasicAndroidCTLogger(BuildConfig.DEBUG))
```
Comment (saran-epifi) [inline core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java:424]: We should only set logger for debug buidls.

Comment (rohan-epifi) [reply core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java]: Done
